### ISSUE TITLE
How to read a subset of a multiband raster?

### ISSUE THREAD

**Issue by bjoern234** [yeesian/ArchGDAL.jl#78]
I've tried to read a subset but with several bands at once from a VRT. The documentation says this can be done using `ArchGDAL.read(dataset, indices, rows, cols)`
But when I do so:
```
    ArchGDAL.registerdrivers() do
        ArchGDAL.read(path_vrt) do dataset
             arr = ArchGDAL.read(dataset, Int32[1,2,3,4], 1:100, 1:100)
        end
    end
```
all columns and rows will be loaded.
What have I done wrong?

**Comment by bjoern234** [comment]
Ahh ok, it seems that output buffer get the size from the whole image
https://github.com/yeesian/ArchGDAL.jl/blob/9cc11c3f3caa57b57680725a356b24db8ff069c8/src/raster/rasterio.jl#L353-L354

**Comment by yeesian** [comment]
Ohh I believe that's a bug, and it should have been something like
```julia
buffer = Array{getdatatype(getband(dataset, indices[1]))}(
    undef, length(cols), length(rows), length(indices)
)
```
like in https://github.com/yeesian/ArchGDAL.jl/blob/9cc11c3f3caa57b57680725a356b24db8ff069c8/src/raster/rasterio.jl#L215 instead

**Comment by yeesian** [comment]
Oh it shows up in other places, like https://github.com/yeesian/ArchGDAL.jl/blob/9cc11c3f3caa57b57680725a356b24db8ff069c8/src/raster/rasterio.jl#L324-L336 too.

We'll have to be careful about how rasterio is invoked (for [`linespace` and `bandspace`](https://github.com/yeesian/ArchGDAL.jl/blob/9cc11c3f3caa57b57680725a356b24db8ff069c8/src/raster/rasterio.jl#L2-L55)) in those cases as well

**Comment by scottstanie** [comment]
Just checking, is there any reason to not do a few quick fixes like change `width(dataset), height(dataset)` to `xsize, ysize` here https://github.com/yeesian/ArchGDAL.jl/blob/9cc11c3f3caa57b57680725a356b24db8ff069c8/src/raster/rasterio.jl#L333 ? when I did that it worked for me (although still a few other bugs, like the band indices had to be `Int32`s)

**Comment by visr** [comment]
Yeah it would be good to have those fixes. If they don't cover all cases, we can keep the issue open for the rest. Would be great if you could submit a PR with a test that shows the new behavior.
